Feature/agenda-module-aesthetic-changes

emilyboda opened the pull request:
1. Moves the time and event titles further to the left, to allow more space for event titles. 
* Events often have long titles that get cut off, especially if users increase font size

2. Right aligns the time
* I feel like it looks better aesthetically to have the end of the time and the beginning of the event be right next to each other. 
* I also added a space before the dot to match the space after

3. Added "all day" to replace a missing time marker
* this may need to be translated, but I'm not sure how to do that. is there a better way to do it than just make a list of the words "all day" in every language? maybe the date module already has this built in?

Aesthetic changes look like this:
https://photos.app.goo.gl/Ti9gbmB6kwJw7Gj6A

emilyboda commented:
apologies, i made the PR too early. still refining the code. will reopen later

emilyboda commented:
Ok, updated the code

aceisace commented on inkycal/modules/inkycal_agenda.py:
```diff
@@ -202,11 +202,15 @@ def generate_image(self):
                               (time_width, line_height), time,
-                              font=self.font, alignment='left')
+                              font=self.font, alignment='right')
+                    if parser.all_day(_):
```
An else statement would be better suited here instead of re-calling the function, causing a little more overhead

aceisace commented:
I like the changes! It looks more compact while the text is well-readable. Thank you for your PR @emilyboda . Let's just wait a little until the tests have passed before attempting to merge

Just pulling your image here so it can be seen even if the link expires or goes down:
<img src="https://github.com/aceinnolab/Inkycal/assets/29558518/8d346ccc-31a0-4d67-a4b1-604c45fb227e">

aceisace commented:
Oh, the tests can't pass as the github secrets aren't configured in your repo. I'll manually run the unittests on my end with your changes

aceisace commented:
I tested your changes locally and it seems there is a small issue with the time-width. The time-width needs some buffer for the width to show "all day". Currently, it will show only "all da":
![image](https://github.com/aceinnolab/Inkycal/assets/29558518/96135288-a15d-4293-b252-c25712dcd4d2)
Setting time-width to:
```py
time_width = int(max([self.font.getlength(
                events['begin'].format(self.time_format, locale=self.language))
                for events in upcoming_events]) + 10)
```
notice the + 10 pixel buffer as this solves this issue:
![image](https://github.com/aceinnolab/Inkycal/assets/29558518/e9b3fbf1-c22f-42f9-a367-8871f87ab5ba)

The red boxes are the boxes generated by the write-function, this time with a red background to better understand the positions.

Please modify the time_width accordingly and then this can be merged 😁

emilyboda commented:
Ok, I fixed it. Rather than adding the arbitrary 10, I added "all day" to the list of times that the program dynamically finds the max of. This should work for "all day" and also it will still work even if that string is changed to another language where the text is longer.

I also found one more unneeded buffer that I removed.

![IMG_5897](https://github.com/aceinnolab/Inkycal/assets/9170143/1917d660-dccf-4402-9c6a-81c8c0a39489)

aceisace commented:
I'm still  getting the same issue, the y on the day is missing. For now, let's settle on the + 10. Later, this can be adjusted once a better version has been found

emilyboda commented:
strange... Ok I added the +10

emilyboda commented:
whoops... NOW i added the plus 10

aceisace commented:
Looks good, thank you very much for your effort in creating this PR and making the agenda more compact, while ensuring readability of the text! Way to go @emilyboda ! Merging this now 👍
